pkg/cloudevents/transport/amqp/transport_test.go

```diff
@@ -0,0 +1,108 @@
+package amqp_test
+
+import (
+	"context"
+	"net/url"
+	"os"
+	"testing"
+
+	"github.com/cloudevents/sdk-go/pkg/binding/spec"
+	"github.com/cloudevents/sdk-go/pkg/binding/test"
+	ce "github.com/cloudevents/sdk-go/pkg/cloudevents"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/transport"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/transport/amqp"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/types"
+	"github.com/stretchr/testify/assert"
+)
+
+// TODO(alanconway) Need better self-test without external dependency.
+
+// Requires an external AMQP broker or router, skip if not available.
+// The env variable TEST_AMQP_URL provides the URL, default is "/test"
+//
+// One option is http://qpid.apache.org/components/dispatch-router/indext.html.
+// It can be installed from source or from RPMs, see https://qpid.apache.org/packages.html
+// Run `qdrouterd` and the tests will work with no further config.
+func testTransport(t *testing.T, opts ...amqp.Option) *amqp.Transport {
+	t.Helper()
+	addr := "test"
+	s := os.Getenv("TEST_AMQP_URL")
```

Comment:
@alanconway  I've also locally updated the example, by applying the dispatch router logic 

```
	t, err := amqp.New(os.Getenv("TEST_AMQP_URL"), "test", amqp.WithEncoding(amqp.StructuredV03))

```

Can you update these examples, similar to the test ? and remove (or change) the `envConfig` struct for sender/receiver in here: https://github.com/cloudevents/sdk-go/tree/master/cmd/samples/amqp

Perhaps worth to sponosr a little `README.md` file, with infos, like in the above test ?

Reply:
Done